// Copyright 2024 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
option cc_enable_arenas = true;

package ray.serve;

option java_package = "io.ray.serve.generated";

message ASGIRequest {
  bytes pickled_request_metadata = 1;
  bytes request_args = 2;
  bytes request_kwargs = 3;
}

message ASGIResponse {
  bytes serialized_message = 1;
}

service ASGIService {
  rpc HandleRequest(ASGIRequest) returns (ASGIResponse);
  rpc HandleRequestStreaming(ASGIRequest) returns (stream ASGIResponse);
  rpc HandleRequestWithRejection(ASGIRequest) returns (stream ASGIResponse);
}
